ProxyAgent/Proxy.cs

```diff
@@ -114,7 +115,7 @@ public class Proxy : IProxy
             catch (RedirectException e)
             {
                 responseOut.StatusCode = (int) e.StatusCode;
-                responseOut.Headers.Add(LOCATION_HEADER, e.Location);
+                responseOut.Headers[LOCATION_HEADER] = e.Location;
```

Comment:
If there is no LOCATION_HEADER, will this add it?

Reply:
this will replace the header value. The Location header doesn't support multiple values because the browser can follow only one location.

Reply:
so we were previously adding a second location header, yes?

Reply:
it was technically possible, but not happening

Reply:
let me revisit the comment thread:
* answering to Jill: yes, the header is added
* answering to Tim: no, we were not adding a second location header

This code runs only when redirecting a client from HTTP to HTTPS, which happens *before* contacting the remote endpoint, thus there are no headers in `responseOut`.
Technically, one could change the code and inject another middleware upfront of this one, and inject a Location header. In that scenario, it's important to *overwrite* the existing Location header, with this one.
So, to summarize, the previous code was ok, but it's more futureproof taking care of the rare edge case.